packages/@remirror/extension-callout/src/callout-types.ts

```diff
@@ -0,0 +1,22 @@
+import type { ProsemirrorAttributes } from '@remirror/core';
+
+/**
+ * Options available to the [[`CalloutExtension`]].
+ */
+export interface CalloutOptions {
+  /**
+   * The default callout type to use when none is provided.
+   *
+   * It is a property so it can change during the editor's life.
+   *
+   * @default 'info'
+   */
+  defaultType?: string;
+}
+
+export interface CalloutAttributes extends ProsemirrorAttributes {
+  /**
+   * The type of callout, for instance info, warning, error or success. Defaults to info
+   */
+  type?: string;
```

Comment:
Was unsure if this should be a `info/warning/error/success` enum, but figured a string might be more flexible for custom types like `critical`?

Reply:
Yeah this makes sense 👍

Reply:
If you want to have the best of both worlds, with support for autocompletion and also the ability to set any string you can use the following from `type-fest`.

```typescript
import { LiteralUnion } from 'type-fest';

export interface CalloutAttributes extends ProsemirrorAttributes {
  /**
   * The type of callout, for instance info, warning, error or success. Defaults to info.
   */
  type?: LiteralUnion<'info' | 'warning' | 'error' | 'success', string>;
}
```